src/mlpack/methods/pca/pca_impl.hpp

```diff
@@ -95,11 +93,11 @@ double PCA<DecompositionPolicy>::Apply(arma::mat& data,
   // Parameter validation.
   if (newDimension == 0)
     Log::Fatal << "PCA::Apply(): newDimension (" << newDimension << ") cannot "
-        << "be zero!" << endl;
+        << "be zero!" << std::endl;
```

Comment:
Since this isn't part of a header file, I don't necessary see the benefit of using the `std::` prefix. I guess sometimes it makes it more clear, but I think for `endl` it's pretty clear. Anyway, I'm not against merging this in, but if we do we should update the style guidelines and use the same style over the codebase.

Reply:
If I am not mistaken, pca.hpp includes pca_imp.hpp

Reply:
Right, what I meant is a user wouldn't include the `pca_impl.hpp`.

Reply:
to be clear, the problem is not the style. The issue is by including pca.hpp in your code, the std namespace is getting used by default. It pollutes the application namespace.

Reply:
Maybe I missed something but you modified `pca_impl.hpp` not `pca.hpp`;  `pca_impl.hpp` is the implementation, `pca.hpp` is the part that is going to be included by an application.

I'm certainly not against the change, but if we do that, let's do it for all the other places as well.

Reply:
respectfully, look at the last line in pca.hpp. It includes pca_imp.hpp:
```
// Include implementation.
#include "pca_impl.hpp"
```

Reply:
So to pollute the application namespace, and we talk here about some application that links against mlpack, not mlpack itself. The  application would include `pca.hpp` but that doesn't mean it can use anything from std because `pca_impl.hpp` uses `using namespace std;`. Maybe I still misunderstood your point.

Reply:
The following example compiles without any error when `pca.hpp` is included:

```
#include <cstdlib>
#include <mlpack/methods/pca/pca.hpp>
#include <string>
#include <iostream>

/*
 * 
 */
int main(int argc, char** argv) {
    
    string str = "Hello world";
    cout << str << endl;
    return 0;
}
```

Whereas the same code without `pca.hpp` included gives the following errors:

```
main.cpp:11:5: error: ‘string’ was not declared in this scope
main.cpp:12:5: error: ‘cout’ was not declared in this scope
main.cpp:12:13: error: ‘str’ was not declared in this scope
main.cpp:12:20: error: ‘endl’ was not declared in this scope
```
As you can see, by including the `pca.hpp` into your code, the std namespace is used by default which is **completely wrong**.

Reply:
Thanks for being persistent, if this works, we should definitely fix this, do you mind to fix the other places as well? I can also run a regex over the codebase and change the necessary places, up to you.

Reply:
I checked the source code and I think that's the only place, but it doesn't hurt if you double check too.

Reply:
Looks like you are right.